cmake/android/OpenCVDetectAndroidSDK.cmake

```diff
@@ -56,8 +56,13 @@ endmacro()
 macro(ocv_detect_android_sdk_tools)
   # https://developer.android.com/studio/releases/sdk-tools.html
   if(NOT DEFINED ANDROID_SDK_TOOLS)
-    if(DEFINED ANDROID_SDK AND EXISTS "${ANDROID_SDK}/tools")
-      set(ANDROID_SDK_TOOLS "${ANDROID_SDK}/tools" CACHE INTERNAL "Android SDK Tools path")
+    if(DEFINED ANDROID_SDK)
+      if (EXISTS "${ANDROID_SDK}/cmdline-tools")
```

Comment:
They are not the same and can't be exchanged. Just compare:

cmdline-tools/latest/source.properties:
```
Pkg.Revision=7.0
Pkg.Path=cmdline-tools;7.0
Pkg.Desc=Android SDK Command-line Tools
```

vs

tools/source.properties
```
Pkg.UserSrc=false
Pkg.Revision=26.1.1
Platform.MinPlatformToolsRev=20
Pkg.Dependencies=emulator
Pkg.Path=tools
Pkg.Desc=Android SDK Tools
```

---

OpenCV doesn't use "tools" with modern "gradle" flow actually. They are use with legacy "ant" build flow.

---

Try to comment out `#ocv_detect_android_sdk_tools()` call.

Reply:
Replaced to a simple check that tools found.

`cmdline-tools` tools changed their versioning so the latest one is 9.0. More about my build steps here: https://github.com/opencv/opencv/pull/23642